packages/ngx-flicking/projects/ngx-flicking/src/lib/ngx-flicking.component.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2015 NAVER Corp.
+ * egjs projects are licensed under the MIT license
+ */
+import NativeFlicking, { Plugin, FlickingOptions, withFlickingMethods, DEFAULT_OPTIONS, FlickingEvent } from '@egjs/flicking';
+// tslint:disable-next-line: max-line-length
+import { Component, OnInit, Input, AfterViewInit, ElementRef, OnChanges, Output, EventEmitter, OnDestroy, ContentChild, TemplateRef } from '@angular/core';
+import ListDiffer, { DiffResult } from '@egjs/list-differ';
+
+@Component({
+  selector: 'eg-flicking',
+  template: `
+    <div>
+      <div class={{classPrefix}}-viewport>
+        <div class={{classPrefix}}-camera ngxChildrenDiffer (update)="onUpdate($event)">
+          <ng-container *ngTemplateOutlet="template"></ng-container>
+          <ng-container *ngFor="let items of [].constructor(cloneCount)">
+            <ng-container *ngTemplateOutlet="template"></ng-container>
+          </ng-container>
+        </div>
+      </div>
+    </div>
+  `,
+  styles: [
+    ':host {display: block}', ':host > div {width: 100%; height: 100%}'
+  ]
+})
+export class NgxFlickingComponent implements OnInit, AfterViewInit, OnDestroy, OnChanges {
+  @Input() options: Partial<FlickingOptions> = {};
+  @Input() plugins: Plugin[] = [];
+
+  // Reference Flicking.EventType
+  @Output() handleEvents = new EventEmitter<Partial<FlickingEvent>>();
+  @ContentChild(TemplateRef) template: TemplateRef<any>;
+
+  @withFlickingMethods
+  private flicking?: NativeFlicking | null;
+  private pluginsDiffer: ListDiffer<Plugin> = new ListDiffer<Plugin>();
```

Comment:
@jongmoon 
I think it's better not to use private accessors in a consistent way.

Reply:
@daybrush 
`classPrefix` and `cloneCount` are used in `html template`. So It need not to use private. But others is not used on others. I think It should be private if it is intended as public.